controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -568,76 +575,76 @@ private boolean verifyRecordMatchesInput(List<Long> segmentsToSeal, List<SimpleE
     }
 
     @Override
-    public CompletableFuture<Void> scaleCreateNewSegments(boolean isManualScale) {
+    public CompletableFuture<VersionedMetadata<EpochTransitionRecord>> startScale(boolean isManualScale,
+                                                                                  VersionedMetadata<EpochTransitionRecord> record,
+                                                                                  VersionedMetadata<State> state) {
+        return getHistoryIndexFromStore().thenCompose(historyIndex -> getHistoryTableFromStore()
+                .thenCompose(historyTable -> getSegmentIndexFromStore().thenCompose(segmentIndex -> getSegmentTableFromStore()
+                        .thenCompose(segmentTable -> {
+                            if (isManualScale) {
+                                return migrateManualScaleToNewEpoch(record, state, historyIndex, historyTable, segmentIndex, segmentTable);
+                            } else {
+                                // if rolling transactions happened before the scale could be picked up, we will discard
```

Comment:
I'm not sure I understand the situation described here and why that would cause the epoch transition record to be dropped. Does it mean that we need to start over the scale operation?

Reply:
> I'm not sure I understand the situation described here and why that would cause the epoch transition record to be dropped. 
Note: there are two workflows that change the epoch of a stream - scaling and rolling transaction. 

Epoch transition is created before the scale processing is attempted. 
First logical step in scale processing is to set the state to SCALING. 
If between having set the epoch transition the state to SCALING, some other processor came and performed a rolling transaction then the epoch transition may become inconsistent with the metadata state. 
Note: because we had an epoch transition record, no scaling could have happened to have changed the epochs. Its only one or more rolling transcations. And rolling transaction end up with exact duplicate of the active epoch from the time when ETR was successfully created. 
So a manual scale can still work with the duplicate of the active epoch and we migrate the ETR as such to the latest duplicate epoch.
We dont perform the migration for an auto scale request (though we could have done the same). We discard the ETR right here. 

>Does it mean that we need to start over the scale operation?
the manual scale operation is migrated to new set of segments. 
the auto scale operation is discarded because the segments it wanted sealed were already sealed because of rolling txn.